lib/image_render.dart

```diff
@@ -90,14 +90,17 @@ ImageRender assetImageRender({
 
 ImageRender networkImageRender({
   Map<String, String> headers,
+  String baseUrl,
```

Comment:
I think I'm going to disagree with you here on this one... It seems like this ought to be a thing of composition rather than adding an extra property

Reply:
That would be nice, yes. Not sure how to achieve that. 

Maybe by extracting the (ugly, but necessary) logic with the FutureBuilder that awaits the first frame to be loaded (to know the size). But I'm not sure how much help that would be.

Reply:
Or maybe a wrapper that passes a mocked new set of attributes to the real implementation, having prefixed the base url? Let me try that.

Reply:
Oh, actually, I see the issue here......

Reply:
I was thinking something of the form:

```dart
ImageRender relativeRender({<normal properties>}, baseURL) => { networkImageRender(<normal properties>, url: baseURL + relativePath) }

```

Reply:
But that also seems a bit silly.... I see the problem here

Reply:
Uhhhhh. Well.... What's your preference here? I'm thinking I'm good either way

Reply:
I guess we could to a rewriteUrl function instead of the baseUrl?
```dart
final src = urlRewriter?.call(_src(attributes)) ?? _src(attributes);
```
Which allows you to rewrite any url, including prefixing with a baseUrl:
```dart
networkImageRender(urlRewriter: (url) => "https://upload.wikimedia.org" + url),
```
How does that sound?

Reply:
That looks okay to me? I think the big fight was between the composition of functions vs a property.

If you feel it would be more useful to have a function that seems good to me. Maybe rename it `mapUrl`? instead?

Reply:
I updated the name. I certainly agree having yet another property isn't ideal, but at least with the mapper function it is powerful enough to prevent even more properties in the future now. For example you can swap domains or force https schema. For me it is fine as-is now.